cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/web/ACSearchRunner.java

```diff
@@ -287,23 +267,88 @@ private void search(Subject subject)
         }
     }
 
-    private void writeError(SyncOutput syncOutput, int code, Throwable t)
-    {
-        try
-        {
-            syncOutput.setResponseCode(code);
+    private void writeError(SyncOutput syncOutput, int code, Throwable t) {
+        try {
+            syncOutput.setCode(code);
             syncOut.setHeader("Content-Type", "text/plain");
             OutputStream ostream = syncOut.getOutputStream();
-            if (ostream != null)
-            {
+            if (ostream != null) {
                 OutputStreamWriter w = new OutputStreamWriter(ostream);
                 w.write(t.toString());
                 w.flush();
             }
-        }
-        catch (IOException e)
-        {
+        } catch (IOException e) {
             log.warn("Could not write response to output stream", e);
         }
     }
+
+    /**
+     * Validate the incoming parameters.
+     * 
+     * IVOA GMS requests can be:
+     *   - (no params) - list group memberships
+     *   - group=[groupName] - is member of {groupName}?
+     * CADC legacy requests can be:
+     *   - role=[role] - list groups in role (member, admin, or owner)
+     *   - role=[role] groupID=[groupName}] - list group if user has specified role in group
+     * 
```

Comment:
I'd like to propose a change in how this is described plus one functional change that should feed back to the WD:

I don't think "CADC legacy" is the right way to characterise this. The OpenCADC implementation supports additional roles and a `role` parameter to refine the query. That's an extended feature set, not legacy. IVOA standards should always allow implementations to have more features. We should document the groupID param as legacy and change our clients to use the standard param (and eventually phase it out of server side).

Functional change: I can see several scenarios where a service has a small number of groups that are allowed and is weighing "N isMember calls" vs "1 getMemberships call". If the group/groupID param had multiplicity 0..* then it could be "1 isMember call with a small list"... it avoids the worst case scenario with a user with too many groups. I have a TODO comment in half a dozen pieces of code with this question, and multiple groupID params would solve it. This comes from operational experience and I don't think this is a tough sell...and it's easy to implement here.

Reply:
I really like this idea.  I've changed the API to take 0..* group params.  I've changed the description of the 'legacy' scenarios to be the 'CADC extension of GMS'.